src/operator/mkl/mkl_relu-inl.h

```diff
@@ -207,23 +212,30 @@ class MKLReluOp : public Operator {
     Tensor<xpu, 4, DType> m_out_data;
     Tensor<xpu, 4, DType> m_in_grad;
     Tensor<xpu, 4, DType> m_in_data;
+
     if (out_grad[activation::kOut].ndim() == 2) {
       Shape<4> dshape = Shape4(out_grad[activation::kOut].shape_[0],
                                out_grad[activation::kOut].shape_[1], 1, 1);
-      m_out_grad = out_grad[activation::kOut].get_with_shape_direct<xpu, 4, DType>(dshape, s);
-      m_out_data = out_data[activation::kOut].get_with_shape_direct<xpu, 4, DType>(dshape, s);
-      m_in_grad = in_grad[activation::kData].get_with_shape_direct<xpu, 4, DType>(dshape, s);
+      mkl_set_priv_flag(out_grad[activation::kOut]);
+      m_out_grad = out_grad[activation::kOut].get_with_shape<xpu, 4, DType>(dshape, s);
+      mkl_set_priv_flag(out_data[activation::kOut]);
+      m_out_data = out_data[activation::kOut].get_with_shape<xpu, 4, DType>(dshape, s);
+      mkl_set_priv_flag(in_grad[activation::kData]);
+      m_in_grad = in_grad[activation::kData].get_with_shape<xpu, 4, DType>(dshape, s);
     } else {
-      m_out_grad = out_grad[activation::kOut].get_direct<xpu, 4, DType>(s);
-      m_out_data = out_data[activation::kOut].get_direct<xpu, 4, DType>(s);
-      m_in_grad = in_grad[activation::kData].get_direct<xpu, 4, DType>(s);
+      mkl_set_priv_flag(out_grad[activation::kOut]);
```

Comment:
let's do this without interface change when not using experimental.
instead of 
```
mkl_set_priv_flag(out_grad[activation::kOut]);
m_out_grad = out_grad[activation::kOut].get<xpu, 4, DType>(s);
```
Write a function `mkl_experimental_direct_get` and do
`m_out_grad = mkl_experimental_direct_get(out_grad[activation::kOut], s)`